File: p3.py
x = 1
N = 1

def f(n,y):
	z = (n*(y**2)+1)**(0.5)
	return z

def pfsq(n):				#to check if n is a perfect square.
	sqt = n**(0.5)
	if int(sqt)==sqt:
		return True
	else:
		return False

flist=[]
nlist=[]	
for n in range(2,100000):
	if not pfsq(n):
		y=1
		nlist.append(n)
		while int(f(n,y))!=f(n,y) and n!=61 and n!=109:
			nex = f(n,y+1)
			if int(nex)==nex :
				flist.append(nex)
				print n,
				print nex
			y+=1
print f(2,3)

File: p3.cpp
#include <iostream>
#include <cmath>
using namespace std;

double f(int n,long long int y){
	double z = sqrt((n*(y*y)+1));
	return z;
}

bool pfsq(int n){
	double sqt = sqrt(n);
	if (int(sqt)==sqt) return true;
	else return false;
}

int main(){
	long long int X=1;
	int N=1;
	for(int n=2;n<100000;n++){
		if(!pfsq(n)){
			long long int y=1;
			while(int(f(n,y))!=f(n,y)){
				double nex=f(n,y+1);
				if(int(nex)==nex){
					if(nex>X){X=nex; N=n;}
				}
				y++;
			}
		}
	}
	cout<<N<<endl;
}
